src/shutdown-handler.ts

```diff
@@ -1,10 +1,15 @@
-import { logError, logger } from './helpers';
+import { logError, logger, resolveOrTimeout } from './helpers';
 
 const SHUTDOWN_SIGNALS = ['SIGINT', 'SIGTERM'] as const;
 
 export type ShutdownHandler = () => void | PromiseLike<void>;
+export type ShutdownConfig = {
+  name: string;
+  handler: ShutdownHandler;
+  timeoutHandler: ShutdownHandler | null;
```

Comment:
Would be helpful if this was more clear on what the difference between a `null` value and a function. Does a null value mean the handler shouldn't be time-constricted (because it's a critical routine), or vice versa?

Maybe either a tsdoc/jsdoc comment on the property, or an additional explicit property?